packages/ariakit-utils/src/__tests__/array-test.ts

```diff
@@ -8,3 +13,34 @@ test("toArray", () => {
   expect(toArray({})).toEqual([{}]);
   expect(toArray([])).toEqual([]);
 });
+
+test("addItemToArray", () => {
+  const arr = ["a", "b", "d"];
+  expect(addItemToArray(arr, "c", 2)).toEqual(["a", "b", "c", "d"]);
+  // Empty Array
+  expect(addItemToArray([] as string[], "c", 2)).toEqual(["c"]);
+  // No Index
+  expect(addItemToArray(arr, "c")).toEqual(["a", "b", "d", "c"]);
+  // Index > length
+  expect(addItemToArray(arr, "c", 99)).toEqual(["a", "b", "d", "c"]);
+  // Immunitibility
+  expect(addItemToArray(arr, "c", 2)).not.toEqual(arr);
```

Comment:
`toEqual` doesn't check object reference.
```suggestion
  expect(addItemToArray(arr, "c", 2)).not.toBe(arr);
```